### PR TITLE
[feat][feat] Delta Neutral on Biswap

### PR DESCRIPTION
**Pull request by alpanaca**
## Description
Allow Automated Vault on Biswap

## Why?
More option for investor

## ChangeLogs:
- DeltaNeutralBiswapWorker03
- worker test
- gateway test

### CODE REVIEW

File: contracts/8.10/protocol/workers/DeltaNeutralBiswapWorker03.sol [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+import "../interfaces/ISwapRouter02Like.sol";
+import "../interfaces/IStrategy.sol";
+import "../interfaces/IWorker03.sol";
+import "../interfaces/IDeltaNeutralOracle.sol";
+import "../interfaces/IVault.sol";
+
+import "../../utils/SafeToken.sol";
+import "../../utils/FixedPointMathLib.sol";
+
+/// @title DeltaNeutralBiswapWorker03 is a SpookyWorker03 worker with reinvest-optimized and beneficial vault buyback functionalities
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/workers/DeltaNeutralBiswapWorker03.sol:33]:
```suggestion
/// @title DeltaNeutralBiswapWorker03 is a BiswapWorker03 worker with reinvest-optimized and beneficial vault buyback functionalities
```

File: contracts/8.10/protocol/workers/DeltaNeutralBiswapWorker03.sol [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+  /// @dev Liquidate the given position by converting it to BaseToken and return back to caller.
+  function liquidate(
+    uint256 /*id*/
+  ) external override onlyOperator nonReentrant {
+    // NOTE: this worker does not allow liquidation
+    revert DeltaNeutralBiswapWorker03_NotAllowToLiquidate();
+  }
+
+  /// @dev Some portion of a bounty from reinvest will be sent to beneficialVault to increase the size of totalToken.
+  /// @param _beneficialVaultBounty - The amount of BOO to be swapped to BTOKEN & send back to the Vault.
```
Comment (sora-alpaca) [inline contracts/8.10/protocol/workers/DeltaNeutralBiswapWorker03.sol:308]:
```suggestion
  /// @param _beneficialVaultBounty - The amount of BSW to be swapped to BTOKEN & send back to the Vault.
```

File: test/helpers/deploy.ts [hunk 3]
```diff
@@ -951,6 +953,57 @@ export class DeployHelper {
...
+    await deltaNeutralWorker03.deployed();
+
+    await simpleVaultConfig.setWorker(deltaNeutralWorker03.address, true, true, workFactor, killFactor, true, true);
+    await deltaNeutralWorker03.setStrategyOk(extraStrategies, true);
+    await deltaNeutralWorker03.setReinvestorOk(okReinvestor, true);
+    await deltaNeutralWorker03.setTreasuryConfig(treasuryAddress, reinvestBountyBps);
+
+    extraStrategies.push(addStrat.address);
+    extraStrategies.forEach(async (stratAddress) => {
+      const strat = SpookySwapStrategyLiquidate__factory.connect(stratAddress, this.deployer);
```
Comment (sora-alpaca) [inline test/helpers/deploy.ts:1000]: BiswapStrategyLiquidate ???

**Review by sora-alpaca — CHANGES_REQUESTED** [review]

**Review by sora-alpaca — APPROVED** [review]

File: test/integrations/protocol/workers/delta-neutral/Vault_DeltaNeutralBiswapWorker03.test.ts [hunk 4]
```diff
@@ -0,0 +1,1737 @@
...
+            let [workerLPAfter] = await masterChef.userInfo(POOL_ID, deltaNeutralWorker.address);
+            expect(workerLPAfter.sub(workerLPBefore)).to.eq(parseEther("1.131776307937023350"));
+            // DeltaNet think he made enough. He now wants to close position partially.
+            // He close 50% of his position and return all debt
+            const deltaNetBefore = await baseToken.balanceOf(deltaNetAddress);
+            const [deltaNetHealthBefore] = await vault.positionInfo("1");
+            const lpUnderPosition = await deltaNeutralWorker.totalLpBalance();
+            [workerLPBefore] = await masterChef.userInfo(POOL_ID, deltaNeutralWorker.address);
+            // DeltaNet think he made enough. He now wants to close position partially.
+            // After calling `work()`, the `_reinvest()` is invoked
```
Comment (jr-alpaca) [inline test/integrations/protocol/workers/delta-neutral/Vault_DeltaNeutralBiswapWorker03.test.ts:1288]: work() doesn't  trigger reinvest in delta neutral worker

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**